servers/quarkus-common/src/main/java/org/projectnessie/quarkus/providers/storage/PersistProvider.java

```diff
@@ -137,12 +138,30 @@ public Persist producePersist() {
       info = " (" + info + ")";
     }
 
-    String cacheInfo;
+    CacheSizing.Builder cacheSizing = CacheSizing.builder();
     int cacheCapacityMB = storeConfig.cacheCapacityMB();
     if (cacheCapacityMB > 0) {
-      CacheBackend cacheBackend = PersistCaches.newBackend(1024L * 1024L * cacheCapacityMB);
+      cacheSizing.fixedSizeInMB(cacheCapacityMB);
+    }
+    int cacheFractionMin = storeConfig.cacheCapacityFractionMinSizeMb();
+    if (cacheFractionMin != 0) {
+      cacheSizing.fractionMinSizeMb(cacheFractionMin);
+    }
+    double cacheCapacityFraction = storeConfig.cacheCapacityFractionOfHeap();
+    if (cacheCapacityFraction > 0d) {
+      cacheSizing.fractionOfMaxHeapSize(cacheCapacityFraction);
+    }
+    int cacheCapacityFractionAdjustMB = storeConfig.cacheCapacityFractionAdjustMB();
+    if (cacheCapacityFractionAdjustMB > 0) {
+      cacheSizing.heapSizeAdjustmentMB(cacheCapacityFractionAdjustMB);
+    }
+    int effectiveCacheSizeMB = cacheSizing.build().calculateEffectiveSizeInMB();
```

Comment:
nit: how about `CacheSizing.from(StoreConfig config)`?

Reply:
Nope, it's a quarkus-config